src/components/pages/confirm/membership/index.tsx

```diff
@@ -7,16 +7,50 @@ import Spinner from 'components/spinner'
 import {IconTwitter} from 'components/share'
 import usePurchaseAndPlay from 'hooks/use-purchase-and-play'
 import {Topic} from 'types'
-import {useTheme} from 'next-themes'
 import PostPurchase from 'components/survey/tally/post-purchase'
+import {trpc} from 'trpc/trpc.client'
+import {useAccount} from 'hooks/use-account'
 
 type HeaderProps = {
   heading: React.ReactElement
   primaryMessage: React.ReactElement
 }
 
 type ConfirmMembershipProps = {
-  session: any
+  session_id: string
+}
+
+const LinkToLatestInvoice = () => {
+  const {account} = useAccount()
+  const {data: subscriptionData, status: subscriptionLoadingStatus} =
+    trpc.subscriptionDetails.forStripeCustomerId.useQuery({
+      stripeCustomerId: account?.stripe_customer_id,
+    })
+  const {data: charge, status: chargeLoadingStatus} =
+    trpc.stripe.balanceTransactionById.useQuery({
+      chargeId: subscriptionData?.latestInvoice?.charge as string,
+    })
```

Comment:
I did few calls here: `account` (to get `stripe_customer_id `) --> `subscriptionData` (to get `latestInvoice. charge ` id) --> `charge` (to get actually `balance_transaction`).
There are couple of errors during fetching process (because I don't get data for the next call immediately):
![Image 2023-01-27 at 17 58 04](https://user-images.githubusercontent.com/1519448/215236239-38ef38c3-6b99-4fe4-bd52-dc8e2c24d1b2.jpg)

Not sure how to handle it. Wrap in promises?

Reply:
technically it works:
![charge](https://user-images.githubusercontent.com/1519448/215236287-6229a1d9-9872-4b11-83af-34fe8e55f825.gif)